test/clientScopes.spec.ts

```diff
@@ -0,0 +1,646 @@
+// tslint:disable:no-unused-expression
+import * as chai from 'chai';
+import {KeycloakAdminClient} from '../src/client';
+import {credentials} from './constants';
+import ClientScopeRepresentation from '../src/defs/clientScopeRepresentation';
+import ProtocolMapperRepresentation from '../src/defs/protocolMapperRepresentation';
+import ClientRepresentation from '../src/defs/clientRepresentation';
+
+const expect = chai.expect;
+
+declare module 'mocha' {
+  // tslint:disable-next-line:interface-name
+  interface ISuiteCallbackContext {
+    kcAdminClient?: KeycloakAdminClient;
+    currentClientScope?: ClientScopeRepresentation;
+    currentClientScopeName?: string;
+    currentClient?: ClientRepresentation;
+  }
+}
+
+describe('Client Scopes', () => {
+  before(async () => {
+    this.kcAdminClient = new KeycloakAdminClient();
+    await this.kcAdminClient.auth(credentials);
+  });
+
+  beforeEach(async () => {
+    this.currentClientScopeName = 'best-of-the-bests-scope';
+    await this.kcAdminClient.clientScopes.create({
+      name: this.currentClientScopeName,
+    });
+    this.currentClientScope = await this.kcAdminClient.clientScopes.findOneByName(
+      {
+        name: this.currentClientScopeName,
+      },
+    );
+  });
+
+  afterEach(async () => {
+    // cleanup default client scopes
+    try {
```

Comment:
I use this `afterEach` to cleanup the test environment. Means I remove the client scopes, realm, ... as a test case might fail, I just "try" to remove all the things, because some things might have never been created due to failures.

Reply:
Thanks for the answer and significant contributing! 

I will merge these codes because all of the other parts are ok, but this block will be changed for more visibility because it not a good practice using try-catch with an empty response based that is hiding errors. In this case, it's not super critical right now.

Reply:
Thanks for the review and the final merge!

I am open for changes regarding this part :-). In comparison to the other tests in the library I changed the way how they construct / deconstruct the test environment. So changes are welcome! :-)